app/common/helpers/filter.js

```diff
@@ -22,9 +22,11 @@ export const setFilter = (filter, { location, router }) => {
 export const getFilter = ({ location: { query } }, filters) => {
   const [{ name: defaultFilter }] = filters;
 
-  const filter = Object.keys(query).find(key =>
-    filters.some(({ name }) => name === key)
-  );
+  const filter = filters[0].names
+    ? Object.keys(query).filter(key =>
+        filters[0].names.some(item => item === key)
+      )
+    : Object.keys(query).find(key => filters.some(({ name }) => name === key));
 
   return filter || defaultFilter;
```

Comment:
Need comment here @kiurchv

Reply:
I've rewritten this function:
```js
export const getFilter = ({ location: { query } }, filters) => {
  const [defaultFilter] = filters;

  const filter = filters.find(
    ({ names, name }) =>
      names
        ? names.every(name => query.hasOwnProperty(name))
        : query.hasOwnProperty(name)
  );

  const { name, names } = filter || defaultFilter;

  return names || name;
};
```

Now it looks much better